### DISCUSSION TITLE
Hooking up to TTL video?

### DISCUSSION THREAD

**Discussion by Lee-Adamson** [keirf/FF_OSD#52] [General]
I apologize if this has already been covered elsewhere.  My search-fu was unable to turn up the info I was looking for, though, and so here I am.  ;P

I'll paste below my thoughts on the subject from a VCF forum post I just made.  The original question was just "Have any of you hooked the flashfloppy osd up to a TTL video signal, like TGA/CGA, MDA, or the Kaypro or TRS-80 Model II/3/4?"

But then I started thinking to myself............

-----

The hookup diagram on their wiki shows a 270 ohm resistor in series with the video signal from the bluepill, hooked up to an analogRGB/VGA type display. It looks to me like that resistor in combination with the 75 ohm termination in the monitor would act as a voltage divider to reduce the bluepill's signal voltage to around 0.7v, compatible with VGA/analogRGB. It appears to just assert this signal in tandem with the system video.

When the bluepill asserts low on its video line, the 270 ohm resistor would have to pull the system video signal to ground in parallel with the 75 ohm termination in the monitor. That seems like it would be a really low resistance and more or less short out the system video signal, but I guess analog RGB signals are able to cope with that without something blowing up.

When the bluepill is not drawing its little OSD square, it must tristate its video signal. Otherwise it would pull the whole screen down while it was active, which it does not appear to do in the videos.

Trying to hook up a TTL video signal like that, however, is going to draw waaaaay more current from the system video line than it is probably designed for, though, and eventually damage it. Additionally, the 3.3v signal from the bluepill may appear very dim, depending on how the analog video circuitry is built.

I suspect that the "correct" way to hook the OSD up to a TTL video signal is to combine the system video and the bluepill's with a logic gate, and to buffer the H and V sync signals such that the gate lag is the same across all 3-4 signals (or 6 signals in the case of CGA). I suspect that a 7432 would be sufficient for this task. However, doing it this way would prevent the OSD from blacking out the system video within its little square. If indeed the bluepill is tristating itself when outside of the OSD square, it ought to be possible to modify the firmware to assert some kind of "active" pin instead, which we could use to gate the OSD video signal to the monitor and suppress the system video, which would make it correctly black out the system video in the OSD area.

Does this make any kind of sense, or am I pulling a Dunning-Kruger move here? What percentage full of poo am I with this?

Thanks again.

**Comment by keirf** [comment]
What you suggest makes sense. In fact OSD already has a configurable output mode where the single Tri-state output pin is changed into a pair of push-pull pins: output and output enable. This is to allow the kind of buffering you discuss.

**Comment by Lee-Adamson** [comment]
The project was a success, although I need to add a pot on there to trim the brightness to match the Kaypro's system video.  Although it may just be a matter of the FF-OSD font being so much thicker than the system font that makes it seem so much brighter.

This was accomplished with a 74LS157 2-1 multiplexer.  If you think it would be useful documentation, I shall endeavor to motivate myself to draw a schematic later and post it here.

The LS157, being a quad-type device, can actually support overlay on up to 4 separate TTL video signals.  This should allow this mod to also work with MDA (video+intensity; 2 lines) and CGA (RGBI; 4 lines), displaying full-intensity white rather than a dim primary color.  Although I have not tested it with anything other than Kaypro video as of yet.

![image](https://user-images.githubusercontent.com/89491638/196743187-87cee3bd-d583-4a86-b33c-e53243424f68.png)

![image](https://user-images.githubusercontent.com/89491638/196743431-d5efe1f0-5c87-4074-bd3a-b95ca66406d7.png)
